Prose/ProseLib/Sources/ConversationFeature/Chat/ChatReducer.swift

```diff
@@ -65,150 +80,192 @@ extension JSEventError: CustomDebugStringConvertible {
 public enum ChatAction: Equatable {
   case webViewReady, alertDismissed
   case navigateUp, navigateDown
-  case messageMenuItemTapped(MessageMenu.Action), menuDidClose
+  case messageMenuItemTapped(MessageMenuAction), menuDidClose
   case message(MessageAction)
   case jsEventError(JSEventError)
+  case reactionPicker(ReactionPickerAction), reactionPickerDismissed
 }
 
+// MARK: - Reducer
+
 let chatReducer = Reducer<
   ChatState,
   ChatAction,
   ConversationEnvironment
-> { state, action, environment in
-  switch action {
-  case .webViewReady:
-    state.isWebViewReady = true
-    return .none
-
-  case .alertDismissed:
-    state.alert = nil
-    return .none
-
-  case .navigateUp:
-    if let messageId = state.selectedMessageId,
-       let index = state.messages.index(id: messageId)
-    {
-      if index > 0 {
-        state.selectedMessageId = state.messages[index - 1].id
+>.combine([
+  reactionPickerTogglingReducer._pullback(
```

Comment:
Weird naming. What you have is a multi-selection. Selections usually can be toggled, so that's not really the distinguishing factor. I'd go for `reactionPickerSelectionReducer` or just `reactionPickerReducer` which handles a selection.

Reply:
Without this reducer, a click always selects a reaction. With it, a click selects or deselects a reaction. This feels like a toggle button, so it made sense for me.
WDYT?

Reply:
Alright